controller_test.py

```diff
@@ -0,0 +1,177 @@
+    # test 104
+    input_patcher = mock.patch('builtins.input', return_value = "Mario")
+    input_mock = input_patcher.start()
+    expected_result = generate_player()
+    self.assertEqual(expected_result.name, "Mario")
+    self.assertEqual(input_mock.call_count, 1)
+    self.assertIsInstance(expected_result, Player)
+    input_patcher.stop()
+
+  def test_user_is_prompted_to_enter_their_name(self):
+    # test 105
+    input_patcher = mock.patch('builtins.input', return_value = "Mario")
+    input_mock = input_patcher.start()
+    generate_player()
+    input_mock.assert_called_once_with("Please enter your name: ")
+    input_patcher.stop()
+
+class PrepareListOfPlayersTest(unittest.TestCase):
+  def test_prepare_list_of_players_returns_a_list(self):
+    # test 106
+    input_patcher = mock.patch('builtins.input', return_value = 2)
+    input_mock = input_patcher.start()
+    actual_result = prepare_list_of_players()
+    self.assertIsInstance(actual_result, list)
+    input_patcher.stop()
+
+  def test_prepare_list_of_players_requests_input_of_number_of_players(self):
+    # test 107
+    input_patcher = mock.patch('builtins.input', return_value = 2)
+    input_mock = input_patcher.start()
+    prepare_list_of_players()
+    self.assertEqual(input_mock.call_args_list[0], mock.call(
+      "Please enter the number of players: "))
+    input_patcher.stop()
+
+  @patch('controller.generate_player')
+  def test_for_2_players_generate_player_is_called_twice(
+    self, generate_player_mock):
+    # test 108
+    manager = mock.Mock()
+    manager.attach_mock(generate_player_mock, 'generate_player')
+    input_patcher = mock.patch('builtins.input', return_value = 2)
+    input_mock = input_patcher.start()
+    prepare_list_of_players()
+    self.assertEqual(generate_player_mock.call_count, 2)
+    self.assertEqual(input_mock.call_args_list[0], mock.call(
+      "Please enter the number of players: "))
+    input_patcher.stop()
+
+  @patch('controller.generate_player')
+  def test_for_3_players_generate_player_is_called_three_times(
+    self, generate_player_mock):
+    # test 109
+    manager = mock.Mock()
+    manager.attach_mock(generate_player_mock, 'generate_player')
+    input_patcher = mock.patch('builtins.input', return_value = 3)
+    input_mock = input_patcher.start()
+    prepare_list_of_players()
+    self.assertEqual(input_mock.call_args_list[0], mock.call(
+      "Please enter the number of players: "))
+    self.assertEqual(generate_player_mock.call_count, 3)
+    input_patcher.stop()
+
+class StartNewGameTest(unittest.TestCase):
+  @patch('bites.Bites.__init__')
+  @patch('bites.Bites.play_full_game')
+  def test_start_new_2player_game_calls_prepare_list_of_players_and_associated_functions_in_correct_order_and_with_correct_callcount(
+    self, mock_bites_play, mock_bites_init):
+    # test 110
+    manager = mock.Mock()
+    mock_bites_init.return_value = None
+    manager.attach_mock(mock_bites_play, 'play_full_game')
```

Comment:
No assertions are done on the `manager` so I don't think you need this object.
```suggestion
    mock_bites_init.return_value = None
```